src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -224,6 +224,10 @@ private void SolutionManager_NuGetProjectRenamed(object sender, NuGetProjectEven
             _mcs = await GetServiceAsync(typeof(IMenuCommandService)) as OleMenuCommandService;
             if (null != _mcs)
             {
+                // Switch to Main Thread before calling AddCommand which calls GetService() which should
+                // always be called on UI thread.
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
can GetServiceAsync be called from BG thread?

Reply:
yes